components/script_traits/lib.rs

```diff
@@ -807,3 +811,19 @@ impl ScriptToConstellationChan {
         self.sender.send((self.pipeline_id, msg))
     }
 }
+
+/// A Script to Embedder channel.
+#[derive(Clone, Deserialize, Serialize)]
+pub struct ScriptToEmbedderChan {
```

Comment:
This is absolutely unecessary by the way, since `top_level_browsing_context_id` would available directly at the point where could be used to send an `EmbedderMsg`...